### PR TITLE
generic pop up modal

### PR DESCRIPTION
**Pull request by shahidrightpoint**
This is a generic pop up modal which can be use to show pop up message.
I have added two button to show how we can use this modal for two or more different purposes. I have added one button to show a success message and other one to error message. 
**Example**: Suppose we are submitting a form and after getting response based on success or failure we can initialized or load this modal with contents change.
Screenshot (490)
**when I click on Success Button**
Below are the screenshot of the component and how I show this component by button.
Screenshot (489)
**when I click on Error Button**
Screenshot (491)

### CODE REVIEW

File: Assets/js/popupmodal.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    const okbtn = document.getElementById("cancelBtn");
+    okbtn.addEventListener("click", () => {
+      document.getElementById("myModal").classList.remove("show");
+    })
+
+    // buttons to load the pop up modal
+    document.getElementById("successPopUp").addEventListener("click", () => {
+        document.getElementById("myModal").classList.add("show")
+        document.querySelector(".cmp-custom-modal__content-icon").setAttribute("src", "../../images/success_icon.png");
+        document.querySelector(".cmp-custom-modal__content-title").innerHTML = "Hurrah! Your message sent";
```
Comment (mathrik) [inline Assets/js/popupmodal.js:12]: It's a little unusual to have the javascript write the text for a popup, but it definitely works.

**Review by mathrik — COMMENTED** [review]
This all looks good to me. My only comment would be that we prefer to use rems over px. I'll have @jwetmore-hoodoo take a look as well, but overall this looks solid.

File: Assets/js/popupmodal.js (hunk 1 above)
Comment (shahidrightpoint) [reply Assets/js/popupmodal.js:12]: @mathrik , You are absolutely right, I have set these text only to demonstrate and once it is added as AEM component then all text and images will be set from cq_dialogue and will required small changes. I have added these text through js to show two purposes for this pop up modal, which can be used in your pages or forms.

**Comment by jwetmore-hoodoo** [comment]
@mathrik it really comes down to what makes sense and what's already used in a project. Rems scale better than pixels, but you may not want horizontal padding/margin to scale, but generally you do want font sizes.

There is now a Sass function `toRem(...)` to help with that in this repo, and could replace some/all of the px values.

Additionally, it's a newer technology, but there's now a [dialog](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog) element that has its own JS functions for open/close and include a `::backdrop` pseudo-element. It can help simplify the amount of code we write, but I'll admit I'm not very familiar so it could be more work depending on design.

**Pull request merged**